src/main/scala/strawman/collection/Map.scala

```diff
@@ -0,0 +1,44 @@
+package strawman.collection
+
+import strawman.collection.mutable.Builder
+
+import scala.Option
+import scala.annotation.unchecked.uncheckedVariance
+import scala.Predef.???
+
+/** Base Map type */
+trait Map[K, +V]
+  extends Iterable[(K, V)]
```

Comment:
@dwijnand If `Map` iterates with `Product2`s, then my typical way of iterating over it is not safe any more:
```scala
for ((key, value) <- myMap) { ... }
```
The new way to iterate on the client side would be:
```scala
for (keyValue <- myMap) {
  val key = keyValue._1
  val value = keyValue._2
  ...
}
```
That's definitely a regression to me.

The principle of least power needs to be balanced with the principle of easiest use, sometimes.